### ISSUE TITLE
There are no instructions on how to build this program

### ISSUE THREAD

**Issue by bert8128** [OpenCppCoverage/OpenCppCoverage#26]
## Describe your environment
- Version of OpenCppCoverage:0.9.6.1
- Architecture (x86/ x64): x64
- Windows version: 7

## Describe your problem
Steps to reproduce:
1.Download source
2. Open sln file in visual studio 17
3. Build project

I then get lots of errors, eg:
NuGet Package restore failed for project TestCoverageOptimizedBuild: Unable to find version '1.0.0' of package 'thirdparty'.
  C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\: Package 'thirdparty.1.0.0' is not found on source 'C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\'.
  https://api.nuget.org/v3/index.json: Package 'thirdparty.1.0.0' is not found on source 'https://api.nuget.org/v3/index.json'.
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\IDE\VC\VCTargets\Platforms\x64\PlatformToolsets\v141\Toolset.targets(36,5): error MSB8036: The Windows SDK version 10.0.15063.0 was not found. Install the required version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution".

I tried running the BuildThirdPartyDependencies batch file.  This referes to a folder Build\ThirdParty which does not exist, and consequently generates the error "The system cannot find the path specified.".  Then there is various downloading followed by an authetication error.  Full log:

C:\source\Bertie\Projects2\OpenCppCoverage-master>BuildThirdPartyDependencies.bat
The system cannot find the path specified.
Cloning into 'vcpkg'...
remote: Counting objects: 31982, done.
remote: Compressing objects: 100% (16/16), done.
remote: Total 31982 (delta 5), reused 14 (delta 4), pack-reused 31962
Receiving objects: 100% (31982/31982), 6.40 MiB | 1.70 MiB/s, done.
Resolving deltas: 100% (21111/21111), done.
Checking out files: 100% (2024/2024), done.
Exception calling "DownloadFile" with "2" argument(s): "The remote server returned an error: (407) Proxy Authentication Required."
At C:\source\Bertie\Projects2\OpenCppCoverage-master\vcpkg\scripts\fetchDependency.ps1:63 char:59
+             (New-Object System.Net.WebClient).DownloadFile <<<< ($url, $downloadPath)
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : DotNetMethodException

Some instructions for building would be very useful!

**Comment by bert8128** [comment]
The query is somewhat resolved.  The problem stems from two issues:

1. I am behind a corporate firewall, and the BuildThirdPartyDependencies is constructed in such a way as to fall foul of the proxy.  I normally find that an ntclm server fixes this, but the code in this cases uses tools that do not allow for a local proxy.
2. I did not have powershell in my path.
Since there is no documentation, it is a bit hard to find the problem.
I ahve somewhat resolved the issue by manually executing the installation of the various packages refered to in BuildThirdPartyDependencies, after having manually installed vcpkg, cmake and git, and copied vswhere from the internet into the place expected.

**Comment by OpenCppCoverage** [comment]
Hello,

I am working to fix the compilation because it failed now for Visual Studio 15.5. When done, I already planned to write a Build instruction page.

OpenCppCoverage

**Comment by OpenCppCoverage** [comment]
Hello,

Here is the [wiki](https://github.com/OpenCppCoverage/OpenCppCoverage/wiki/Building-OpenCppCoverage) explaining how to build OpenCppCoverage.

OpenCppCoverage

**Comment by dzolee** [comment]
Small notes (maybe saves time for someone else)

-I was unable to download vswhere, there was a SSL error returned by Powershell. I have installed "chocolatey" and it installed vswhere 

-Also failed to produce vcpkg.exe using the tools, I have manually downloaded it from github, built with bootstrap batch file then copied the vcpkg.exe to the appropriate folder of OpenCppCoverage

-I have build the 3rd party stuff and the Nuget package. Nuget package exported, then I receive the msg from vcpkg:

"With a project open, go to Tools->NuGet Package Manager->Package Manager Console and paste:
    Install-Package thirdparty -Source "D:\OpenCppCoverage\Build\ThirdParty\vcpkg"

I do this but the message in Visual Studio is:
"Install-Package : Some NuGet packages are missing from the solution. The packages need to be restored in order to build the dependency graph. Restore the packages before performing any 
operations."

I try "Restore" in VS but no success...

**Comment by ras0219-msft** [comment]
@dzolee Try removing all existing NuGet packages from the project before running the `Install-Package` command

**Comment by OpenCppCoverage** [comment]
Hello, 

I fixed the third party compilation.
I hope it will help,

OpenCppCoverage

**Comment by dzolee** [comment]
Yes, very good job. After updating the source, I started an Admin PowerShell and launched BuildThirdPartyDependencies.bat. It finished relatively quickly (I had done the compilation previously) and it said it added the package to the project folder successfully.

I tried to build the solution with VS2017, I got an error "cannot find windows.h". It turned out quickly that there was a target SDK mismatch. I have another (a newer) version installed. I have updated all projects and all configs to this value (10.0.16299.0), and it compiled like a charm:

![grafik](https://user-images.githubusercontent.com/10893838/40902348-058978be-67d4-11e8-9fd0-599c70e77320.png)

I will try to "test" the build procedure again by setting up everything from scratch on a new empty Win7 virtual machine. It will take a few days ...
